### PR TITLE
Feature/Adds functionality to pass a timestamp for an event log

### PR DESCRIPTION
**Pull request by willmero**

### CODE REVIEW

File: clientSuccessClient.js [hunk 1]
```diff
@@ -394,8 +395,9 @@ JS.class(ClientSuccessClient, {
 		 * @param  {String}  contactID       - ID of the contact that the activity originated from
 		 * @param  {String}  activity        - Activity name that occurred
 		 * @param  {Integer} [occurrences=1] - Number of times that the user completed this action
+		 * @param  {Date}    [timestamp]     - ISO 8601 formatted timestamp that the event occured
 		 */
-		trackActivity : async function({ clientID = undefined, contactID = undefined, activity = undefined, occurrences = 1 }) {
+		trackActivity : async function({ clientID = undefined, contactID = undefined, activity = undefined, occurrences = 1, timestamp = undefined }) {
```
Comment (bewmar) [inline clientSuccessClient.js]: This object should `= {}` to set as a default

File: clientSuccessClient.js [hunk 2]
```diff
@@ -394,8 +395,9 @@ JS.class(ClientSuccessClient, {
 		 * @param  {String}  contactID       - ID of the contact that the activity originated from
```
Comment (bewmar) [inline clientSuccessClient.js]:
```js
@param {Object} [options]
@param {String} [options.clientID]
...
```

File: clientSuccessClient.js [hunk 3]
```diff
@@ -407,6 +409,9 @@ JS.class(ClientSuccessClient, {
 					},
 				},
 				value : occurrences,
+				keen  : {
+					timestamp : timestamp ? moment(timestamp).toISOString() : '',
```
Comment (bewmar) [inline clientSuccessClient.js]: Prefer `new Moment`

File: clientSuccessClient.js [hunk 4]
```diff
@@ -391,13 +391,14 @@ JS.class(ClientSuccessClient, {
...
+		 * @param  {String}  [options.clientID]      - ID of the ClientSuccess client that the usage will be logged under
+		 * @param  {String}  [options.contactID]     - ID of the contact that the activity originated from
+		 * @param  {String}  [options.activity]      - Activity name that occurred
+		 * @param  {Number}  [options.occurrences=1] - Number of times that the user completed this action
+		 * @param  {Date}    [options.timestamp]     - ISO 8601 formatted timestamp that the event occured
 		 */
-		trackActivity : async function({ clientID = undefined, contactID = undefined, activity = undefined, occurrences = 1, timestamp = undefined }) {
+		trackActivity : async function({ clientID, contactID, activity, occurrences = 1, timestamp } = {}) {
 			this.validateClientSuccessId(clientID);
 			const client   = await this.getClient(clientID);
```
[the remark this answers is not in the document]
Comment (willmero) [reply clientSuccessClient.js]: Sniper

**Pull request merged**